packages/compass-collection/src/components/mock-data-generator-modal/mock-data-generator-modal.tsx

```diff
@@ -83,9 +92,27 @@ const MockDataGeneratorModal = ({
       case MockDataGeneratorStep.PREVIEW_DATA:
         return <></>; // TODO: CLOUDP-333857
       case MockDataGeneratorStep.GENERATE_DATA:
-        return <ScriptScreen />;
+        // UX flow ensures faker schema is completed, but TypeScript needs the guard
+        if (fakerSchemaGenerationState.status === 'completed') {
+          return (
+            <ScriptScreen
```

Comment:
This is a weird split in component responsibilities, feels better suited for the ScriptScreen. So is the property passing, all this stuff is coming from state, so you should just connect the ScriptScreen component to this state directly as the redux documentation recommends